### DISCUSSION TITLE
YCM 0.8.0 Released

### DISCUSSION THREAD

**Discussion by drdanz** [robotology/community#297] [Releases]
YCM 0.8.0 was released today and is now available for download at: https://github.com/robotology/ycm/releases/.

Changes made since YCM 0.6 include the following.

## Build System

* CMake 3.5 or newer is now required.

### Generic Modules

* `GitInfo`: Fixed for clones using `git-worktree`
* `InstallBasicPackageFiles`: Added `INCLUDE_FILE` argument to support appending user content to the generated `PackageConfig.cmake` file.
* `InstallBasicPackageFiles`: Added `EXPORT` arguments to use `export(EXPORT)` instead of `export(TARGETS)` (default = <name>) if `TARGETS`-related arguments are not passed.
* `InstallBasicPackageFiles`: Added `FIRST_TARGET` argument to specify a target to check when using `EXPORT` (default = <name>)
* `InstallBasicPackageFiles`: Added `COMPONENT` argument, forwarded to the `install` commands.
* `InstallBasicPackageFiles`: Added `INSTALL_DESTINATION` and `EXPORT_DESTINATION` arguments, to install and generate the files in a path different from the default. The `DESTINATION` argument is now deprecated in favour of `INSTALL_DESTINATION`.
* `InstallBasicPackageFiles`: Added `PRIVATE_DEPENDENCIES` argument to support dependencies that should be located only if the targets are built `STATIC` (i.e. libraries linked as `PRIVATE`).
* `InstallBasicPackageFiles`: Fixed support for CMake < 3.9.

### Find Modules

* The `FindFFMPEG` was replaced by the 3rd party one.

## CMake Next

* Updated to CMake v3.11.2 + commit Kitware/CMake@f248f8ad7898816a5a627e288d3a04fed993967f.
* Imported modules `FindPython.cmake`, `FindPython/Support.cmake`, `FindPython2.cmake`, and `Find3Python.cmake` from CMake master (3.12).

## 3rd Party

* qt-gstreamer modules are now imported from the official [qt-gstreamer Git Repository](https://github.com/GStreamer/qt-gstreamer/) instead of from the `robotology-dependencies` clone.
* `Catch`, `CatchAddTests`, and `ParseAndAddCatchTests` modules imported from [Catch2 Git Repository](https://github.com/catchorg/Catch2/).
* The `FindFFMPEG` is now imported from the [VTK Git Repository](https://gitlab.kitware.com/vtk/vtk/) (ref v8.1.1).

**Comment by PeterBowman** [comment]
I think the newly added `EXPORT` command to `install_basic_package_files()` is broken, please see https://github.com/robotology/ycm/issues/148.
